### ISSUE TITLE
Allowing both `customDistributionName` and `customDistributionFileNameMapper` would make for a more intuitive API

### ISSUE THREAD

**Issue by tudortimi** [denis-zhdanov/custom-gradle-dist-gradle-plugin#14]
It would be much more intuitive if both `customDistributionName` and `customDistributionFileNameMapper` were allowed.

-----------------------
Here is how it is:

The user sets `customDistributionName` to `"my-company"`. The user is happy with the file name for the custom distribution for a while.

Later on, the user wants to tweak the file name. They have to unset `customDistributionName`. They have to put the custom distribution name somewhere in  `customDistributionFileNameMapper`. They get confused why they can't just use the `distributionName` argument, but this is `null` for a single distribution.

Later on, the user wants to produce multiple distributions.  They have to update `customDistributionFileNameMapper` to now handle the `distributionName` argument.

--------------------------
Here is how it could be:

The user sets `customDistributionName` to `"my-company"`. The user is happy with the file name for the custom distribution for a while.

Later on, the user wants to tweak the file name. They set `customDistributionFileNameMapper`. They get `gradleDist.customDistributionName` passed to the mapper as the `distributionName` argument. The argument is always set, even when there is a single custom distribution.

Later on, the user wants to produce multiple distributions. They can leave `customDistributionFileNameMapper` the same.

The user should get some error if they are setting `customDistributionName` and trying to produce multiple distributions.

**Comment by tudortimi** [comment]
Would mean changing to:

```
fun interface CustomDistributionNameMapper {

    fun map(
        gradleVersion: String,
        customDistributionVersion: String,
        gradleDistributionType: String,
        distributionName: String
    ): CharSequence
}
```

`distributionName` is not nullable anymore.

**Comment by denis-zhdanov** [comment]
fixed in `1.19.0`

**Comment by piotrminkina** [comment]
@denis-zhdanov Are You sure?

The following build scripts are work fine, as just before:

```groovy
plugins {
    id 'tech.harmonysoft.oss.custom-gradle-dist-plugin' version '1.19.0'
}

gradleDist {
    gradleVersion = '8.2.1'
    customDistributionName = 'custom-name'
}
```

```
import tech.harmonysoft.oss.gradle.dist.config.CustomDistributionNameMapper

plugins {
    id 'tech.harmonysoft.oss.custom-gradle-dist-plugin' version '1.19.0'
}

gradleDist {
    gradleVersion = '8.2.1'
    customDistributionFileNameMapper = (CustomDistributionNameMapper) { gradleVersion, customDistributionVersion, distributionType, distributionName ->
        if (distributionName) {
            "gradle-${gradleVersion}-devsecops${customDistributionVersion}-${distributionName}-${distributionType}.zip"
        } else {
            "gradle-${gradleVersion}-devsecops${customDistributionVersion}-${distributionType}.zip"
        }
    }
}
```

But using both properties at the same time causes an error as below, and from what I understand from the issue content, @tudortimi asks to be able to use both properties at the same time. I support this request.

```
> the both 'customDistributionName' and 'customDistributionFileNameMapper' are defined, but only one of them is required
```
